### PR TITLE
Code coverage

### PR DESCRIPTION
**Pull request by eolivi**
I'm not totally sure on how to test this PR. because, to be effective it has to be on master openmeeg.
so let's try like this ? or should I do a project on codecov for eolivi/openmeeg first ? what do you think ?

### CODE REVIEW

**Comment by agramfort** [comment]
FYI I activated codecov.io  https://github.com/organizations/openmeeg/settings/installations

**Comment by papadop** [comment]
OK for me, even though I do not see where the coverage is triggered...

File: build_tools/travis/after_success.sh [hunk 1]
```diff
@@ -14,6 +14,12 @@ function setup_conda {
     source activate testenv
 }
 
+if [[ "$USE_COVERAGE" == "1" ]]; then
+    lcov --directory . --capture --output-file coverage.info # capture coverage info
+    lcov --remove coverage.info '/usr/*' --output-file coverage.info # filter out system
+    lcov --list coverage.info
+    bash <(curl -s https://codecov.io/bash)
+fi
```
Comment (eolivi) [inline build_tools/travis/after_success.sh]: here @papadop

**Comment by eolivi** [comment]
it's is there https://codecov.io/gh/openmeeg/openmeeg/tree/master/OpenMEEG

**Comment by agramfort** [comment]
it's not there yet. @eolivi I guess you need to push again to trigger the
codecov

**Comment by agramfort** [comment]
nice !

**Comment by agramfort** [comment]
now the game is to increase coverage by removing dead code or adding
more tests :)

**Comment by eolivi** [comment]
yes ! not that bad anyway

File: build_tools/travis/after_success.sh (hunk 1 above)
Comment (papadop) [inline build_tools/travis/after_success.sh]: No I was looking for the compiler options. But I guess cmake handles them transparently.

**Comment by agramfort** [comment]
yes. For example it's obvious now that StimInternalPotential is not tested

https://codecov.io/gh/openmeeg/openmeeg/src/master/OpenMEEG/src/gain.cpp

File: build_tools/travis/after_success.sh (hunk 1 above)
Comment (eolivi) [inline build_tools/travis/after_success.sh]: it was already there:
````
cmake/TestOption.cmake:if (ENABLE_COVERAGE)

cmake/TestOption.cmake:        set(COVERAGE_FLAGS "-fprofile-arcs -ftest-coverage")

cmake/TestOption.cmake:        set(CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} ${COVERAGE_FLAGS} -lgcov")

cmake/TestOption.cmake:        set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} ${COVERAGE_FLAGS}")

cmake/TestOption.cmake: 
set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} ${COVERAGE_FLAGS}")

cmake/TestOption.cmake:        message(SEND_ERROR "Coverage is only available with gcc.")

```

**Comment by agramfort** [comment]
+1 for MRG

let's increase the coverage with new PRs

**Comment by papadop** [comment]
Note that it will be difficult to attain 100% code coverage as there are quite a few lines of code related to mathematical libraries that are #if'ed. But the more the better....

**Comment by papadop** [comment]
Merged !

**Pull request merged**